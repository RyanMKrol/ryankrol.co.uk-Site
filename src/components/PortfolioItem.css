.PortfolioItem {
  display: flex;
  text-align: left;
  align-items: center;

  padding: 8px;

  min-height: 200px;
  border-bottom-style: dashed;
  border-bottom-width: 1px;
  border-bottom-color: #CCC;
}

.portfolio-item-description {
  display: flex;
  vertical-align: middle;

  flex: 1;
  padding: 0 8px;
}
.portfolio-item-image {
  display: flex;
  vertical-align: middle;

  min-width: 35%;
  padding: 0 8px;
}
