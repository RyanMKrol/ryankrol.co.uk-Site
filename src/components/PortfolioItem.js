import React, { Component } from 'react'

import './PortfolioItem.css';

class PortfolioItem extends Component {
  constructor(props) {
    super()

    this.portfolioMedia = props.portfolioMedia
    this.portfolioDescription = props.portfolioDescription
    this.portfolioHeader = props.portfolioHeader
    this.portfolioItemLink = props.portfolioItemLink
  }

  render() {
    return (
      <div className="PortfolioItem">
        <div className="portfolio-item-header">
          <h3>
            {this.portfolioHeader}
          </h3>
        </div>
        <div className="portfolio-item-content">
          <div className="portfolio-item-description">
            {this.portfolioDescription}
          </div>
          <div className="portfolio-item-image">
            {this.portfolioMedia}
          </div>
        </div>
        <div className="portfolio-project-link">
          <span className="portfolio-link">Check out the source code <a href={this.portfolioItemLink}>here</a>!</span>
        </div>
      </div>
    )
  }
}

export default PortfolioItem
