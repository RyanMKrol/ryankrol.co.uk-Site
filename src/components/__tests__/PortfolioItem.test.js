import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import PortfolioItem from './../PortfolioItem'

const defaultArgs = {
  portfolioHeader: "Example Item",
  portfolioItemLinkHeader: "http://example.com",
  portfolioDescription: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Arcu dui vivamus arcu felis bibendum ut tristique et egestas. Sed nisi lacus sed viverra tellus. Morbi tincidunt ornare massa eget egestas. Pharetra magna ac placerat vestibulum. Et molestie ac feugiat sed lectus vestibulum mattis. Dolor sit amet consectetur adipiscing elit pellentesque habitant morbi tristique. Non pulvinar neque laoreet suspendisse interdum consectetur. Tempus quam pellentesque nec nam aliquam sem et. Nisl nisi scelerisque eu ultrices vitae auctor eu augue. Sed faucibus turpis in eu. Habitant morbi tristique senectus et netus et malesuada fames ac.",
  portfolioMedia: (<p>
    Text that will eventually be an image
  </p>)
}

// Render Tests

it('renders PortfolioItem correctly', () => {
    const portfolioItem = shallow(
      <PortfolioItem {...defaultArgs}/>)
    expect(toJson(portfolioItem)).toMatchSnapshot()
})

it('renders PortfolioItem without throwing', () => {
    expect(() => mount(<PortfolioItem {...defaultArgs}/>)).not.toThrow()
})
