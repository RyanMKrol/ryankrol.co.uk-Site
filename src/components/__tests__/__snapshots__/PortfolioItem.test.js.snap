// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PortfolioItem correctly 1`] = `
<div
  className="PortfolioItem"
>
  <div
    className="portfolio-item-header"
  >
    <h3>
      Example Item
    </h3>
  </div>
  <div
    className="portfolio-item-content"
  >
    <div
      className="portfolio-item-description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Arcu dui vivamus arcu felis bibendum ut tristique et egestas. Sed nisi lacus sed viverra tellus. Morbi tincidunt ornare massa eget egestas. Pharetra magna ac placerat vestibulum. Et molestie ac feugiat sed lectus vestibulum mattis. Dolor sit amet consectetur adipiscing elit pellentesque habitant morbi tristique. Non pulvinar neque laoreet suspendisse interdum consectetur. Tempus quam pellentesque nec nam aliquam sem et. Nisl nisi scelerisque eu ultrices vitae auctor eu augue. Sed faucibus turpis in eu. Habitant morbi tristique senectus et netus et malesuada fames ac.
    </div>
    <div
      className="portfolio-item-image"
    >
      <p>
        Text that will eventually be an image
      </p>
    </div>
  </div>
  <div
    className="portfolio-project-link"
  >
    <span
      className="portfolio-link"
    >
      Check out the source code 
      <a>
        here
      </a>
      !
    </span>
  </div>
</div>
`;
